lib/puppet/node/environment.rb

```diff
@@ -214,15 +214,22 @@ def validation_errors
     errors
   end
 
+  def rich_data_from_env_conf
+    unless @checked_conf_for_rich_data
+      environment_conf = Puppet.lookup(:environments).get_conf(name)
+      @rich_data_from_conf = (environment_conf.nil? ? nil : environment_conf.rich_data)
```

Comment:
is it time to start using `&.`?

Reply:
Ooh, you're right, I think we can now, looks like that was added in 2.3.0. I'm not used to seeing it so I never think of it...

Reply:
Rubocop is yelling about this, but it's also testing Ruby 2.1 which seems unnecessary...

Reply:
Maybe add `TargetRubyVersion` to .rubocop.yml to match the oldest supported ruby version?

Reply:
I've added a commit that targets Ruby 2.4. In order to target 2.5, we'd have to update Rubocop past 1.0, due to dep requirements in `rubocop-i18n`. I tried going to latest, but the diff is pretty big. I might keep working on that, but this commit fixes this PR for now.